packages/plugin-devtools/test/devtools.test.ts

```diff
@@ -0,0 +1,735 @@
+                '<span class="type">VNode</span> ROOT<span class="id">0</span>' +
+                '</div>';
+            expect(about.outerHTML).to.equal(aResult);
+
+            const properties = devtools.querySelector('devtools-info .properties');
+            const pResult =
+                '<div class="properties">' +
+                '<div class="divider">👤 About me</div>' +
+                '<table>' +
+                '<tbody>' +
+                '<tr><td>id</td><td>0</td></tr>' +
+                '<tr><td>type</td><td>ROOT</td></tr>' +
+                '<tr><td>length</td><td>4</td></tr>' +
+                '<tr><td>total length</td><td>22</td></tr>' +
+                '<tr><td>original tag</td><td>None</td></tr>' +
+                '<tr><td>text</td><td>"Titleabcdefghidiv"</td></tr>' +
+                '</tbody>' +
+                '</table>' +
+                '<div class="divider">👪 My Family</div>' +
+                '<table>' +
+                '<tbody>' +
+                '<tr><td>index</td><td>0</td></tr>' +
+                '<tr><td>children</td><td><ol><li>HEADING1</li><li>PARAGRAPH</li><li>PARAGRAPH</li><li>?</li></ol></td></tr>' +
+                '</tbody>' +
+                '</table>' +
+                '</div>';
+            expect(properties.outerHTML).to.equal(pResult);
+        });
+        it('should display the root in bottom path bar', async () => {
+            const path = devtools.querySelector('devtools-path');
+            expect(path.childNodes.length).to.equal(1);
+            expect(path.firstChild.textContent).to.equal('root');
+            expect((path.firstChild as HTMLElement).classList.contains('selected')).to.equal(
+                true,
+                'button is selected',
+            );
+        });
+    });
+    describe('TreeComponent', () => {
+        it('should select the first paragraph', async () => {
+            const node = devtools.querySelector(
+                'devtools-node .children devtools-node:nth-child(2)',
+            );
+            const name = node.querySelector('.element-name');
+            const pos = name.getBoundingClientRect();
+            name.dispatchEvent(
+                new MouseEvent('click', {
+                    clientX: pos.left + 30,
+                    clientY: pos.top,
+                    bubbles: true,
+                }),
+            );
+            await longNextTick();
+            expect(name.classList.contains('selected')).to.equal(true, 'node is selected');
+
+            expect(node.classList.contains('folded')).to.equal(true, 'node is folded');
+            expect(node.querySelector('.children').children.length).to.equal(
+                0,
+                'children are not loaded',
+            );
+
+            const about = devtools.querySelector('devtools-info .about');
+            const aResult =
+                '<div class="about">' +
+                '<span class="type">VNode</span> PARAGRAPH<span class="id">9</span>' +
```

Comment:
For example, if we change how VNode ids are generated, all tests doing this will fail...